data-science-onramp/data-cleaning/clean.py

```diff
@@ -0,0 +1,179 @@
+import os
+import sys
+import re
+import datetime
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import UserDefinedFunction, lit
+from pyspark.sql.types import StringType, IntegerType, FloatType
+
+
+PROJECT_ID = sys.argv[1]
+BUCKET_NAME = sys.argv[2]
+TABLE = f'{PROJECT_ID}.new_york_citibike_trips.RAW_DATA'
+
+def trip_duration(duration):
+    '''Convert trip duration to seconds.'''
+    if duration:
+        number = re.match('\d*.\d', duration)
+        if number:
+            number = float(number[0])
+        else:
+            return None
+
+        if 's' in duration:
+            return int(number)
+        elif 'm' in duration:
+            return int(number * 60)
+        elif 'h' in duration:
+            return int(number * 60 * 60)
+        else:
+            if number < 0:
+                return None
+            else:
+                return int(number)
```

Comment:
You can remove a few conditionals by switching to something similar to the following:

```
# a negative duration, regardless, should probably be cleaned out
if not number or number < 0: 
    return None

number = float(number[0])

if 'm' in duration:
    number *= 60
elif 'h' in duration:
    number *= 60 * 60

return int(number)

```

Reply:
May want to use a more explicit variable name than `number` as well